classes/mail/Mailable.inc.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+/**
+ * @file classes/mail/Mailable.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Mailable
+ * @ingroup mail
+ *
+ * @brief Represents email message data
+ */
+
+namespace PKP\mail;
+
+use BadMethodCallException;
+use Illuminate\Mail\Mailable as IlluminateMailable;
+use InvalidArgumentException;
+use PKP\context\Context;
+use PKP\core\PKPApplication;
+use PKP\core\PKPContainer;
+use PKP\mail\variables\ContextEmailVariable;
+use PKP\mail\variables\QueuedPaymentEmailVariable;
+use PKP\mail\variables\RecipientEmailVariable;
+use PKP\mail\variables\ReviewAssignmentEmailVariable;
+use PKP\mail\variables\SenderEmailVariable;
+use PKP\mail\variables\SiteEmailVariable;
+use PKP\mail\variables\StageAssignmentEmailVariable;
+use PKP\mail\variables\SubmissionEmailVariable;
+use PKP\payment\QueuedPayment;
+use PKP\site\Site;
+use PKP\stageAssignment\StageAssignment;
+use PKP\submission\PKPSubmission;
+use PKP\submission\reviewAssignment\ReviewAssignment;
+use PKP\user\User;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionParameter;
+
+class Mailable extends IlluminateMailable
+{
+
+    /**
+     * Message object key in data array
+     * @var string
+     */
+    const DATA_KEY_MESSAGE = 'message';
```

Comment:
Laravel injects a variable called `message` into all templates. I prefer to handle it before compilation to avoid unexpected coercions with our variables. I moved the name of this variable into the const.